src/udisks2monitor.cpp

```diff
@@ -422,9 +422,6 @@ void UDisks2::Monitor::updatePartitionStatus(const UDisks2::Job *job, bool succe
                     partition->activeState = operation == UDisks2::Job::Mount ? QStringLiteral("active") : QStringLiteral("inactive");
                     partition->status = operation == UDisks2::Job::Mount ? Partition::Mounted : Partition::Unmounted;
                 }
-            } else {
```

Comment:
Now the failure case goes to UI

Reply:
So if the operations start from the class here, I think also the success is unnecessary since the callback code in startMountOperation() sets the status on success case. startLuksOperation() in similar way having statuses set on the callback.

But then I suppose the code here tries to also follow udisk jobs started by anyone in which case we can't rely on any callback existing and removing handling here would break things.

Reply:
Ok. What is better solution for you?

Reply:
> Don't think there's anything in udisks to correct.
> 

I mean corrections in `startMountOperation()`

> 
> and have the temporary states mounting/unmounting etc override the state by following udisk jobs but only while the jobs are active.

What do you mean?

Reply:
> 
> To rephrase: the code in this project would follow the state that's reported by udisks partition set (mounted/unmounted, that is the end states). It would also separately follow the active job(s) and keep track of their status (mounting/unmounting kind of temporary states). The status getter would be kind of "if (there is active job) return job_status; else return the_partition_status;
> 

1) Do you mean partition->status in `updatePartitionStatus()` as "status getter"?
2) The job can be completed with success = false, but partition status would be Unmouned when Unmount was impossible